### ISSUE TITLE
[Feature Request] Delete grouping concept without contained concepts

### ISSUE THREAD

**Issue by HansBosma** [archimatetool/archi#1070]
### Description

In a view I have Archimate elements that contain other archimate elements. I'd like to delete the grouping concept from the view but not the contained elements. E.g. I have a visual group that contains several applications; I'd like to delete the visual group from the view but not the applications within the group. I think it is not possible to do this, except for dragging the contained elements out of the grouping concept and then delete the group. But that is quite a hassle...

**Comment by RaimondB** [comment]
Agree that it is quite a hassle. A jArchi should be able to solve it, but I would think it should be a standard feature to delete an object in general while keeping the nested objects in the view. It should then work for all types of elements.

**Comment by jbsarrodie** [comment]
Hi,

I agree that this should, one day, become a built-in feature (I've needed it more than one time).

But because a jArchi implementation is possible, 
and there's only Phil working on code (I now seldom find time to code), this won't be implemented anytime soon.

Regards

**Comment by HansBosma** [comment]
ok, appreciate your support. I will do without it... and hope for this feature to be built in, remaining a happy Archi user.

**Comment by romualdrichard** [comment]
I tried a jArchi script but it's just a try :( not working

**Comment by Phillipus** [comment]
> I tried a jArchi script but it's just a try :( not working

Because re-parenting [is not implemented](https://github.com/archimatetool/archi-scripting-plugin/issues/116) in jArchi.

I'm working on a native implementation. When I get it working I'll update here.

**Comment by Phillipus** [comment]
I've implemented this now. It's a generic implementation that will delete the selected container object from the View and re-parent the object's child objects to its immediate parent, which might be another container object or the View (diagram).

All that's left to do is to decide what to name the function in the menu. At the moment we have "Delete from View" and "Delete from Model". The new feature could be something like:

- "Delete from View keeping children"
- "Delete Container from View"
- Something else

Suggestions for a short yet descriptive menu item name are welcome...

**Comment by RaimondB** [comment]
Maybe "delete and keep nested"?

**Comment by romualdrichard** [comment]
"Remove container"
"Delete and ungroup"

I asked Ckaude : 
"Remove parent keep child"
"Delete parent save nested"
"Erase parent keep content"
"Remove box keep contents"
"Delete group keep items"
"Ungroup keep inner objects"
"Remove parent free child"
"Delete shell keep inside"
"Remove wrapper keep inner"
"Erase parent free nested"

**Comment by RaimondB** [comment]
Btw, will it also work with multi select and do we have the function in jarchi as well for enabling automated cleanup scenario's in diagrams

**Comment by Phillipus** [comment]
> Btw, will it also work with multi select and do we have the function in jarchi as well for enabling automated cleanup scenario's in diagrams

Yes, multi-select is supported. A jArchi implementation will come later once this has proved to be OK.

**Comment by Phillipus** [comment]
I still can't decide what to call it. The important words are "delete" and "from View". I tried "Delete Container from View" but not everyone will understand "Container". Perhaps "Delete from View (keep children)"?

**Comment by HansBosma** [comment]
My first idea is to retain the current menu functions but in case it is a container object, ask the user: 'do you want to keep the contained objects in the view?' This question can also be asked whenever one wants to delete a container from the model.

Otherwise, I agree, 'Delete from View (keep children)' or 'Delete from View (keep contained objects'). And also? 'Delete from Model (keep contained objects in the view).

**Comment by Phillipus** [comment]
> My first idea is to retain the current menu functions but in case it is a container object, ask the user: 'do you want to keep the contained objects in the view?' This question can also be asked whenever one wants to delete a container from the model.

I agree that would have been one way to do it, but because of the way things work internally with the Eclipse framework the regular "Delete" command can't be over-ridden to ask that question. That's why we also have a separate "Delete from Model" command. One advantage of having separate commands is that you can assign different shortcut keys.

**Comment by HansBosma** [comment]
ok yes, I agree "Delete from View (keep children)" looks best

**Comment by Phillipus** [comment]
This is in the Archi 5.5 Early Access builds.

**Comment by Phillipus** [comment]
Implemented in Archi 5.5.

**Comment by HansBosma** [comment]
Thank you very much!! I will check the feature, and will be an even happier
user.

Op wo 29 jan 2025 om 14:22 schreef Phil Beauvoir ***@***.***>:
